### ISSUE TITLE
 No 'Access-Control-Allow-Origin' header is present on the requested resource

### ISSUE THREAD

**Issue by vgavrilovikj** [jamesmoriarty/react-instagram-authless-feed#19]
Hello, I have installed the package and included in my Home component like this:
`<Feed userName="jamespaulmoriarty" className="Feed" classNameLoading="Loading" limit="3"/>`

However I keep on getting this issue, not sure how it can be resolved...
![image](https://user-images.githubusercontent.com/29601489/110255370-97e1a680-7f93-11eb-9063-bc962e536cb0.png)

Can you please help me? :)

**Comment by vgavrilovikj** [comment]
@jamesmoriarty

**Comment by jamesmoriarty** [comment]
@vgavrilovikj please upgrade the library to v2.0.0. A change on instagrams side broke previous versions.

**Comment by vgavrilovikj** [comment]
![image](https://user-images.githubusercontent.com/29601489/110256578-e5611200-7f99-11eb-83f7-05f4421eeaa2.png)
this is what I have in my package.json - I believe it's the v2.0.0 but even with this it's not working.. @jamesmoriarty

**Comment by jamesmoriarty** [comment]
Sounds like this: https://github.com/jamesmoriarty/react-instagram-authless-feed/issues/17#issuecomment-791021158

What are you serving your site with?

**Comment by vgavrilovikj** [comment]
@jamesmoriarty  I am using Laravel & reactjs... yeah it does sound like that but I didn't found a solution on that issue.. any idea how to solve this?

**Comment by jamesmoriarty** [comment]
@vgavrilovikj: https://stackoverflow.com/questions/39429462/adding-access-control-allow-origin-header-response-in-laravel-5-3-passport

**Comment by vgavrilovikj** [comment]
@jamesmoriarty tried that but it doesn't work.. I guess it may be, because in Laravel you have CSRF token, and you should use the CSRF token when making an API call, but probably the package doesn't use it, so it may be because of that.. but I really am not sure... as from what I saw so far all those issues related to the headers for Laravel were for "/api/*" routes and not external ones...

**Comment by vgavrilovikj** [comment]
@jamesmoriarty from what I've been trying so far I guess it's something that should be configured in the fetch function in Instagram.js

**Comment by jamesmoriarty** [comment]
Hey, it's on your side. Your laravel application serves a response with headers that restrict clientside requests to instagram.com and other origins. You need to read up on Laravel and CORS. It's to protect your site from cross site scripting.

**Comment by vgavrilovikj** [comment]
@jamesmoriarty  Okay and how can I solve it...? Because even if I add X-CSRF-TOKEN in the fetch () function it still doesn't work..

**Comment by jamesmoriarty** [comment]
Start with this: https://www.twilio.com/blog/handling-cross-origin-resource-sharing-cors-requests-laravel-7

**Comment by vgavrilovikj** [comment]
@jamesmoriarty yeah even installed the package https://github.com/fruitcake/laravel-cors .. and set it up correctly, but still no success sadly... and when making the fetch added this.. 
![image](https://user-images.githubusercontent.com/29601489/110260672-7ee5ef00-7fad-11eb-8f83-7989ffb2e9ca.png) 

but still nothing..

**Comment by vgavrilovikj** [comment]
@jamesmoriarty  I tried adding the header Access-Control-Allow-Origin "*" to the nginx server directly, but still not a success

**Comment by jamesmoriarty** [comment]
It needs to be a response header on the page that makes the request clientside. E.g. index.php returned with the header and the client/browser requests instagram.com.

**Comment by jamesmoriarty** [comment]
Apologies @vgavrilovikj looks like I'm seeing the same thing now. Appears something changed on the remotes this component depends on again.

**Comment by jamesmoriarty** [comment]
Looks like the instagram.com proxy URLs responds inconsistently. I'll try adding in a retry to compensate for this.
![Screenshot 2021-03-08 203904](https://user-images.githubusercontent.com/96487/110303618-b0fb5d80-804e-11eb-82da-2589839c3d3c.png)
![Screenshot 2021-03-08 204043](https://user-images.githubusercontent.com/96487/110303630-b5277b00-804e-11eb-901f-bd89c22ef9b6.png)

**Comment by vgavrilovikj** [comment]
@jamesmoriarty any idea how we can solve this..? Cause I really need this package for a website, as I haven't found any other good way to implement an Instagram feed on a website..

**Comment by vgavrilovikj** [comment]
@jamesmoriarty this is what I have in the response headers: 
![image](https://user-images.githubusercontent.com/29601489/110364699-62e15c80-8044-11eb-9db6-20975adec314.png)

**Comment by vgavrilovikj** [comment]
@jamesmoriarty also I just tried this one, when I open the console I am getting an error but somehow it works..

Here's the link to this package:
https://www.sowecms.com/demos/InstagramFeed/

![image](https://user-images.githubusercontent.com/29601489/110371223-f9198080-804c-11eb-9f3f-7296ab58c7f8.png)

**Comment by jamesmoriarty** [comment]
@vgavrilovikj I'm currently testing this: https://github.com/jamesmoriarty/react-instagram-authless-feed/pull/20/files

Having some issues verifying the fix/replicating the issue.

**Comment by jamesmoriarty** [comment]
Looks like they made the workaround unusable even with the retry now.

<img width="633" alt="Screen Shot 2021-03-18 at 5 45 00 pm" src="https://user-images.githubusercontent.com/96487/111584249-b9535580-8811-11eb-9854-caa297378c00.png">

**Comment by vgavrilovikj** [comment]
Were you able to resolve this...? @jamesmoriarty

**Comment by jamesmoriarty** [comment]
@vgavrilovikj no. Instagram/Google made further changes which broke the last workaround. Hoping a new workaround appears in one of the other Github repositories.
